Source/Api/Controllers/EventController.cs

```diff
@@ -439,6 +443,41 @@ public class EventController : RepositoryApiController<IEventRepository, Persist
         }
 
         /// <summary>
+        /// Suvbmit heartbeat
+        /// </summary>
+        /// <param name="projectId">The identifier of the project.</param>
+        /// <param name="version">The api version that should be used.</param>
+        /// <param name="sessionIdOrUserId">The session id or user id.</param>
+        /// <response code="202">Accepted</response>
+        /// <response code="400">No project id specified and no default project was found.</response>
+        /// <response code="404">No project was found.</response>
+        [HttpGet]
+        [Route("~/api/v{version:int=2}/events/session/{sessionIdOrUserId:minlength(1)}/heartbeat")]
+        [Route("~/api/v{version:int=2}/projects/{projectId:objectid}/events/session/{sessionIdOrUserId:minlength(1)}/heartbeat")]
+        [OverrideAuthorization]
+        [Authorize(Roles = AuthorizationRoles.Client)]
+        public async Task<IHttpActionResult> GetSubmitHeartbeat(string projectId = null, int version = 2, string sessionIdOrUserId = null) {
```

Comment:
I think a better name would be SendHeartbeat or RecordHeartbeat.  Get seems wrong because it's not getting anything. But we are making it a GET to just keep the implementation simple. Doesn't mean the method name has to be Get though.